apiary.apib

```diff
@@ -36,44 +37,79 @@ current session status.
-                    "links": {
-                        "related": "https://example.com/session/account"
-                    },
-                    "data": {
-                        "id": "abc1234",
-                        "type": "account"
+    + Headers
+
+            Location: https://example.com/session
+
+    + Body
+
+            {
+                "links": {
+                    "self": "https://example.com/session"
+                },
+                "data": {
+                    "id": "sessionid123",
+                    "type": "session",
+                    "relationships": {
+                        "account": {
+                            "links": {
+                                "related": "https://example.com/session/account"
+                            },
+                            "data": {
+                                "id": "abc1234",
+                                "type": "account"
+                            }
+                        }
                     }
                 },
-                "profile": {
-                    "links": {
-                        "related": "https://example.com/session/account/profile"
+                "included": [{
+                    "id": "abc1234",
+                    "type": "account",
+                    "attributes": {
+                        "username": "jane-doe"
                     },
-                    "data": {
-                        "id": "profile",
-                        "type": "accountProfile"
+                    "relationships": {
+                        "profile": {
+                            "links": {
+                                "related": "https://example.com/session/account/profile"
+                            },
+                            "data": {
+                                "id": "abc1234-profile",
+                                "type": "accountProfile"
+                            }
+                        }
                     }
-                }
+                }]
             }
-        }
 
 + Response 401 (application/vnd.api+json)
 
         {
-          errors: [{
-            "statusCode": "401",
-            "error": "Unauthorized",
-            "message": "Invalid Credentials"
-          }]
+            errors: [{
+                "status": "401",
+                "title": "Unauthorized",
+                "detail": "Invalid Credentials"
+            }]
+        }
+
++ Response 409 (application/vnd.api+json)
+
+        {
+            errors: [{
+                "status": "409",
+                "title": "Conflict",
+                "detail": "Invalid type provided, only supported type is 'session'"
+            }]
+        }
+
++ Response 403 (application/vnd.api+json)
+
+        {
+            errors: [{
+                "status": "403",
+                "title": "Forbidden",
+                "detail": "There is an active session, log out of the current session before logging in again"
```

Comment:
I think we should not allow to send an `Authorization` header for `PUT /session`. If it is sent, the error should not mention that there is an active session, but that "Authorization header not allowed in `PUT /session`". I think that'd make it more clear.

Because technically, nothing speaks against having multiple sessions at the same time, if that is what your app requires.